geopandas/io/sql.py

```diff
@@ -1,46 +1,70 @@
-import binascii
+import functools
 
-from pandas import read_sql
+import pandas as pd
 import shapely.wkb
 
 from geopandas import GeoSeries, GeoDataFrame
 
 
-def read_postgis(sql, con, geom_col='geom', crs=None, index_col=None,
-                 coerce_float=True, params=None):
+def read_sql(sql, con, geom_col='geom', crs=None, hex_encoded=True,
+             index_col=None, coerce_float=True, params=None):
     """
     Returns a GeoDataFrame corresponding to the result of the query
     string, which must contain a geometry column.
 
-    Examples:
-    sql = "SELECT geom, kind FROM polygons;"
-    df = geopandas.read_postgis(sql, con)
-
     Parameters
     ----------
-    sql: string
-    con: DB connection object or SQLAlchemy engine
-    geom_col: string, default 'geom'
+    sql : string
+        SQL query to execute in selecting entries from database.
+    con : DB connection object or SQLAlchemy engine
+        Active connection to the database to query.
+    geom_col : string, default 'geom'
         column name to convert to shapely geometries
-    crs: optional
-        CRS to use for the returned GeoDataFrame
+    crs : dict or str, optional
+        CRS to use for the returned GeoDataFrame; if not set, tries to
+        determine CRS from the SRID associated with the first geometry in
+        the database, and assigns that to all geometries.
+    hex_encoded : bool, optional
+        Whether the geometry is in a hex-encoded string. Default is True,
+        standard for postGIS. Use hex_encoded=False for sqlite databases.
 
     See the documentation for pandas.read_sql for further explanation
     of the following parameters:
     index_col, coerce_float, params
 
+    Returns
+    -------
+    geodataframe : GeoDataFrame
+
+    Example
+    -------
+    >>> sql = "SELECT geom, kind FROM polygons;"
+    >>> df = geopandas.read_postgis(sql, con)
     """
-    df = read_sql(sql, con, index_col=index_col, coerce_float=coerce_float,
-                  params=params)
 
-    if geom_col not in df:
-        raise ValueError("Query missing geometry column '{0}'".format(
-            geom_col))
+    df = pd.read_sql(sql, con, index_col=index_col, coerce_float=coerce_float,
+                     params=params)
 
-    wkb_geoms = df[geom_col]
+    if geom_col not in df:
+        raise ValueError("Query missing geometry column '{}'".format(geom_col))
 
-    s = wkb_geoms.apply(lambda x: shapely.wkb.loads(binascii.unhexlify(x.encode())))
+    def load_geom(x):
+        if isinstance(x, bytes):
+            return shapely.wkb.loads(x, hex=hex_encoded)
+        else:
+            return shapely.wkb.loads(str(x), hex=hex_encoded)
+    geoms = df[geom_col].apply(load_geom)
+    df[geom_col] = GeoSeries(geoms)
```

Comment:
The GeoSeries(..) here is not strictly needed, as it is only in  the end that it is converted to a GeoDataFrame